sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/aad/webapp/AADHandleConditionalAccessFilter.java

```diff
@@ -0,0 +1,99 @@
+ * will throw {@link WebClientResponseException}, Web APP can handle this exception to challenge the user.
+ */
+public class AADHandleConditionalAccessFilter extends OncePerRequestFilter {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(AADOAuth2OboAuthorizedClientRepository.class);
+
+    @Override
+    protected void doFilterInternal(HttpServletRequest request, HttpServletResponse response,
+                                    FilterChain filterChain) throws IOException, ServletException {
+        try {
+            filterChain.doFilter(request, response);
+        } catch (Exception exception) {
+            WebClientResponseException webClientResponseException =
+                Optional.of(exception)
+                        .map(Throwable::getCause)
+                        .filter(e -> e instanceof WebClientResponseException)
+                        .map(e -> (WebClientResponseException) e)
+                        .filter(AADHandleConditionalAccessFilter::isConditionalAccessExceptionFromObo)
+                        .orElse(null);
+            if (webClientResponseException != null) {
+                handleConditionalAccess(webClientResponseException, request, response);
+                return;
+            }
+            throw exception;
+        }
+    }
+
+    private static boolean isConditionalAccessExceptionFromObo(WebClientResponseException exception) {
+        String result = Optional.of(exception)
+                                .map(WebClientResponseException::getHeaders)
+                                .map(httpHeaders -> httpHeaders.get(HttpHeaders.WWW_AUTHENTICATE))
+                                .map(list -> list.get(0))
+                                .filter(value -> value.contains(Constants.CONDITIONAL_ACCESS_POLICY_CLAIMS))
+                                .orElse(null);
+        return result != null;
+    }
+
+    static void handleConditionalAccess(WebClientResponseException exception, HttpServletRequest request,
+                                        HttpServletResponse response) {
+        Map<String, String> authParameters =
+            Optional.of(exception)
+                    .map(WebClientResponseException::getHeaders)
+                    .map(httpHeaders -> httpHeaders.get(HttpHeaders.WWW_AUTHENTICATE))
+                    .map(list -> list.get(0))
+                    .map(AADHandleConditionalAccessFilter::parseAuthParameters)
+                    .orElse(null);
+        request.getSession().setAttribute(Constants.CONDITIONAL_ACCESS_POLICY_CLAIMS,
+            authParameters.get(Constants.CONDITIONAL_ACCESS_POLICY_CLAIMS));
+        response.setStatus(302);
+        try {
+            response.sendRedirect(Constants.DEFAULT_AUTHORITY_ENDPOINT_URI);
+        } catch (IOException e) {
+            LOGGER.error("Failed to redirect at this response.", exception);
+        }
+    }
+
+    private static Map<String, String> parseAuthParameters(String wwwAuthenticateHeader) {
```

Comment:
Does this wwwAuthenticateHeader's format same to the format we discussed in Teams group? For example: starts with `Bearer `

Reply:
Done.